packages/dbt/sodasql/dbt.py

```diff
@@ -0,0 +1,90 @@
+"""DBT integeration"""
+
+from __future__ import annotations
+
+from typing import Any
+
+from dbt.contracts.graph.compiled import ParsedModelNode, CompiledSchemaTestNode
+from dbt.contracts.results import RunResultOutput
+from dbt.node_types import NodeType
+
+
+def parse_manifest(
+    manifest: dict[str, Any]
+) -> tuple[dict[str, ParsedModelNode], dict[str, CompiledSchemaTestNode]]:
+    """
+    Parse the manifest.
+
+    Only V3 manifest is supported.
+
+    Parameters
+    ----------
+    run_results : dict[str, Any]
+        The raw manifest.
+
+    Returns
+    -------
+    out : tuple[dict[str, ParsedModelNode], dict[str, CompiledSchemaTestNode]]
+        The parsed manifest.
+
+    Raises
+    ------
+    NotImplementedError :
+        If the dbt schema is not equal to the V3 manifest
+
+    Source
+    ------
+    https://docs.getdbt.com/reference/artifacts/manifest-json
+    """
+    dbt_v3_schema = "https://schemas.getdbt.com/dbt/manifest/v3.json"
+    if manifest["metadata"]["dbt_schema_version"] != dbt_v3_schema:
+        raise NotImplementedError("Dbt manifest parsing only supported for V3 schema.")
```

Comment:
Maybe it's better to assert than raising NotImplementedError

Reply:
Typically `raise` is used for user facing checks and `assert` for internal checks. The manifest and run_results are provided by the user, so I would keep a `raise`